### DISCUSSION TITLE
Frontend url as variable

### DISCUSSION THREAD

**Discussion by federicocappellotto97** [directus/directus#9591] [Feature Requests]
It would be nice to be able to use project url as a variable.
I have a presentation link 'preview' with url `http://localhost:3000/preview....`, and of course when this website will go live it will change the url. Instead of changing every field, it would be easier write url like this `${FRONTEND_URL}/preview....`

**Comment by joselcvarela** [comment]
Hello @federicocappellotto97 
You should set `PUBLIC_URL` with your app address (in `.env` file for example), then you can read it from `process.env` on Node.js or `import.meta.env.PUBLIC_URL` on Vue side

**Reply by federicocappellotto97** [reply]
Do you mean like this?
<img width="841" alt="Schermata 2021-11-08 alle 16 02 59" src="https://user-images.githubusercontent.com/63153191/140765748-f17fa364-947a-4635-b8ab-d307b3de00bf.png">

**Reply by joselcvarela** [reply]
Ohhh you mean in the fields... I thought you were asking in code.
That would not work.

My recommendation is to create a custom syntax, for example, `{{PUBLIC_URL}}` and then you use a custom hook to replace this string by `process.env.PUBLIC_URL` for example.

**Reply by CallMeLaNN** [reply]
If `$PUBLIC_URL` [Dynamic Variables](https://docs.directus.io/configuration/filter-rules/#dynamic-variables) have built-in support, html/markdown editor can generate proper (without `localhost:8055`) link during image upload as well as preview image correctly. e.g. `![alt]($PUBLIC_URL/assets/cdf8a0c1-d7c6-4b8f-91d1-e37d74737006)`

**Reply by rijkvanzanten** [reply]
@CallMeLaNN That'd be a fairly heavy operation to do though, as we'd have to find and replace that string in every possible text column in the database on read 🤔

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
